src/components/Pager.vue

```diff
@@ -0,0 +1,374 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2024 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+<template>
+  <div class="pager">
+    <template v-if="pages.length === 1">
+      <slot name="page" :page="pages[0]" />
+    </template>
+    <template v-else>
+      <div class="container">
+        <Gutter class="left">
+          <ControlPrevious
+            :disabled="!hasPreviousPage"
+            @click.native="previous"
+          />
+        </Gutter>
+        <div class="viewport" ref="viewport">
+          <div
+            v-for="({ page, key }, n) in keyedPages"
+            ref="pages"
+            :class="['page', pageStates(n)]"
+            :id="key"
+            :key="key"
+          >
+            <slot name="page" :page="page" />
+          </div>
+        </div>
+        <Gutter class="right">
+          <ControlNext
+            :disabled="!hasNextPage"
+            @click.native="next"
+          />
+        </Gutter>
+      </div>
+      <div class="compact-controls">
+        <ControlPrevious
+          :disabled="!hasPreviousPage"
+          @click.native="previous"
+        />
+        <ControlNext
+          :disabled="!hasNextPage"
+          @click.native="next"
+        />
+      </div>
+      <div class="indicators">
+        <a
```

Comment:
Links without a text description need to have an aria-label describing what they are for.
For this indicator link, let's add an aria-label saying `Show slide [page] of [total length of pages]`

Reply:
https://github.com/apple/swift-docc-render/pull/782/commits/d64f8c58616f6100e368cc9404e2c477d9fbc4b5

I added labels for the indicators `aria-current` to the active label, but I don't think it's appropriate to add `aria-disabled` to the inactive indicators, since they can still be used to navigate to those pages and aren't technically "disabled" in that sense.

Reply:
I mentioned it because I read it at https://www.smashingmagazine.com/2023/02/guide-building-accessible-carousels/ and I thought it made sense. But it would be okay if we don't add it too, it's a bit controversial in my opinion.

> Do not just indicate visually but also semantically which control does represent the currently displayed slide. You can use aria-disabled="true" for this purpose. In this case, aria-disabled is preferable to the HTML attribute disabled. Unlike disabled, a button with an aria-disabled attribute is still included in the tab sequence of the page. You may also use the [aria-current](https://www.w3.org/TR/wai-aria-1.1/#aria-current) property set to "true".